### ISSUE TITLE
Inconsistent virtualenv import messages

### ISSUE THREAD

**Issue by hugorodgerbrown** [Microsoft/pyright#41]
**Describe the bug**

This may related to #30 

Python packages installed in a virtualenv are showing as import errors ("Import 'foo' could not be resolved"), and yet intellisense and tooltips are displaying correctly. This results in 1000s of errors in the project.

**To Reproduce**

Prerequisites: both `python.venvPath` and `venv` settings have been set.

1. Install `pyenv` and `pipenv`
2. Install `Django` into the virtualenv using `pipenv install django`.
3. Open a module in a project that imports Django modules
4. Check the import statements for errors, and 
5. Check specific modules / functions imported from those packages

**Expected behavior**

Packages installed within the virtualenv should not display an error (red squiggly underline), and Pyright should recognise them.

**Actual behavior**

The top-level packages are _not_ recognised, but functions imported from those packages are (or appear to be).

**Screenshots or Code**

Django is installed in the local virtualenv, but `django.utils.encoding` cannot be resolved.

<img width="718" alt="Screenshot 2019-03-31 at 15 33 44" src="https://user-images.githubusercontent.com/200944/55290423-1bb7af00-53cb-11e9-9dcc-4c6b88f2d870.png">

However, `smart_text`, a function within the package _can_ be resolved - as it displays tooltips:

<img width="1032" alt="Screenshot 2019-03-31 at 15 33 51" src="https://user-images.githubusercontent.com/200944/55290438-46a20300-53cb-11e9-8487-fb2884af40f0.png">

And intellisense is working on the packages:

<img width="480" alt="Screenshot 2019-03-31 at 15 53 35" src="https://user-images.githubusercontent.com/200944/55290627-836ef980-53cd-11e9-9040-ecf7fd029552.png">

Screenshot showing that VS Code is using the correct virtualenv environment:

<img width="270" alt="Screenshot 2019-03-31 at 15 45 25" src="https://user-images.githubusercontent.com/200944/55290513-14dd6c00-53cc-11e9-8455-c1717ee40d24.png">

**VS Code extension or command-line**
Running the VS Code extension

**Comment by erictraut** [comment]
You appear to have the standard VS Code Python extension installed alongside Pyright. That's fine — they'll work together, but that's why you're seeing the apparent inconsistent behavior. The Python extension is apparently resolving the imports but Pyright is not. I'll need a little more information to help diagnose why.

First, which version of Pyright are you running? Is it the latest (1.0.6)? I fixed a bug that was in earlier versions that could explain this behavior.

Second, if you open the Output window (View -> Output) and select "Python", you should see some debugging output from Pyright. Does it contain any lines that refer to the pythonPath or venvPath setting?

**Comment by hugorodgerbrown** [comment]
Thanks @erictraut - I've now updated my settings to add an explicit `pyrightconfig.json` file (not just me VSCode settings), and the packages are now resolving, but the functions are not - i.e. it's now the opposite problem (I've disabled the standard Python extension):

<img width="763" alt="Screenshot 2019-03-31 at 16 17 12" src="https://user-images.githubusercontent.com/200944/55290859-77d10200-53d0-11e9-8501-624c91fa5edd.png">


I have taken a look at the Output - and yes, there is some logging:

```shell
Pyright language server starting
Received updated settings.
Loading configuration file at /Users/hugorodgerbrown/Projects/.../pyrightconfig.json
pythonPath /Users/hugorodgerbrown/.local/share/virtualenvs/foo/bin/python is not a valid directory.
Finding source files
Found 1130 source files
/Users/hugorodgerbrown/Projects/... opened.
```

This is my (truncated) `pyrightconfig.json`:

```json
{
  "venvPath": "/Users/hugorodgerbrown/.local/share/virtualenvs",
  "venv": "foo",
  "pythonVersion": "3.7"
}
```

It looks like pyright is expecting `$(venvPath)/bin/python` to be a directory, but it's not, it's an executable:

```
$ ls -al /Users/hugorodgerbrown/.local/share/virtualenvs/bin
-rwxr-xr-x   1 hugorodgerbrown  staff  3325088 24 Mar 12:31 python
-rwxr-xr-x   1 hugorodgerbrown  staff     2383 24 Mar 12:31 python-config
lrwxr-xr-x   1 hugorodgerbrown  staff        6 24 Mar 12:31 python3 -> python
lrwxr-xr-x   1 hugorodgerbrown  staff        6 24 Mar 12:31 python3.7 -> python
```

**Comment by hugorodgerbrown** [comment]
I'm using 1.06 btw.

**Comment by erictraut** [comment]
I've needed to rethink my strategy around import resolution in Pyright. There are a couple of issues. 

First, the VS Code setting "python.pythonPath" is not the same as PYTHONPATH. This generates a lot of confusion, and I was also confused by it. The "python.pythonPath" points to a python interpreter binary, which is of no use to Pyright. The PYTHONPATH environment variable refers to the search paths that a Python interpreter should be used for external imports.

I personally always use virtual environments and never use PYTHONPATH. The latter is too fragile and doesn't work well for projects that involve multiple developers (because inevitably every developer has their PYTHONPATH configured differently). That said, I know that many people do use PYTHONPATH.

I've added support for the PYTHONPATH environment variable in Pyright. Note that if you change the value of this environment variable, you'll need to relaunch VS Code and Pyright before the new value is noticed.

I've also added some [new documentation](https://github.com/Microsoft/pyright/blob/master/docs/import-resolution.md) that discusses Pyright's import resolution behavior and the importance of using type stub files for all imported libraries. Without these type stubs, Pyright will have no type information for imported symbols.

These changes will be in the upcoming 1.0.7 release.
